community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -1089,11 +1038,36 @@ public boolean next()
 
                 try
                 {
-                    return getRecord( currentId, record, mode, pageCursor, pageIdForRecord( currentId ) ).inUse();
+                    try
+                    {
+                        record.setId( currentId );
+                        long pageId = pageIdForRecord( currentId );
+                        if ( pageId == pageCursor.getCurrentPageId() ||
+                                pageCursor.next( pageIdForRecord( currentId ) ) )
+                        {
+                            int offset = offsetForId( currentId );
```

Comment:
If you place the calls to offsetForId and pageIdForRecord, and both inline, then the JIT can extract both the modulo for the offset and the division for the page id, from the same divide instruction.